sdk/resourcemanager/Azure.ResourceManager/src/ResourceIdentifier.cs

```diff
@@ -80,9 +80,12 @@ private void Init(ResourceIdentifier parent, ResourceType resourceType, string n
 
             if (resourceType == Subscription.ResourceType)
             {
-                Guid output;
-                if (!Guid.TryParse(name, out output))
-                    throw new ArgumentOutOfRangeException("resourceId", "Invalid resource id.");
+                if (!string.IsNullOrEmpty(name))
+                {
+                    Guid output;
+                    if (!Guid.TryParse(name, out output))
+                        throw new ArgumentOutOfRangeException("resourceId", "Invalid resource id.");
```

Comment:
maybe we should also use `nameof(resourceId)` instead of string literal here?

Reply:
`resourceId` is not a local variable here. So we have to use `"resourceId"` instead.